rs/p2p/quic_transport/src/request_handler.rs

```diff
@@ -217,14 +217,19 @@ async fn handle_uni_stream(
     request.extensions_mut().insert::<NodeId>(peer_id);
     request.extensions_mut().insert::<ConnId>(conn_id);
 
+    let response = router.oneshot(request).await.expect("Infallible");
     // Record application level errors.
-    if !router
-        .oneshot(request)
-        .await
-        .expect("Infallible")
-        .status()
-        .is_success()
-    {
+    if !response.status().is_success() {
+        let status = response.status();
+        let error = axum::body::to_bytes(response.into_body(), usize::MAX)
+            .await
+            .map(|b| String::from_utf8_lossy(&b).to_string())
+            .unwrap_or("Failed to get error string".to_string());
+
+        error!(
+            log,
+            "Application error for uni stream: status {} error {}", status, error
+        );
```

Comment:
Could it be good add `every_n_seconds` here?

Reply:
It only happens very rarely and I need to think about it how we can avoid it entierly.